experimental/traffic-portal/src/app/api/testing/cache-group.service.ts

```diff
@@ -584,4 +593,39 @@ export class CacheGroupService {
 		}
 		return this.regions.splice(index, 1)[0];
 	}
+
+	public async getASNs(): Promise<Array<ResponseASN>>;
+	public async getASNs(id: number): Promise<ResponseASN>;
+
+	/**
+	 * Gets an array of ASNs from Traffic Ops.
+	 *
+	 * @param id If given, returns only the asn with the given id (number).
+	 * @returns An Array of ASNs objects - or a single ASN object if 'id'
+	 * was given.
+	 */
+	public async getASNs(id?: number): Promise<Array<ResponseASN> | ResponseASN> {
+		if(id) {
+			const asn = this.asns.find(a=>a.asn === id);
```

Comment:
This gets the ASN with the `asn` that equals the given ID, not the ASN with the `id` that matches the given ID, which I believe is incorrect.

Reply:
Ok, let me double conform.